### ISSUE TITLE
FNX3-14783 ⁃ Build fails

### ISSUE THREAD

**Issue by bleonard252** [mozilla-mobile/fenix#1163]
## Steps to reproduce
Build Fenix according to README.md instructions

### Expected behavior
Outputs artifacts

### Actual behavior
Build fails with error

### Device information
* Android device: N/A
* Fenix version: Latest master
* **Using [GitLab](https://gitlab.com/bleonard252/phoenix/-/jobs/182962358) to build**

[Full build log](https://gitlab.com/bleonard252/phoenix/-/jobs/182962358/raw)

**Comment by pocmo** [comment]
Breaking Nightly builds too:
https://tools.taskcluster.net/groups/NHKTWg4TSqWNAOQe2GGXsA/tasks/VTo0YvvdTR6d9zM08OtO9A/runs/0/logs/public%2Flogs%2Flive.log

**Comment by pocmo** [comment]
CC @boek: Caused by the metrics integration.

**Comment by pocmo** [comment]
Looks like it's related to:
https://github.com/mozilla-mobile/android-components/pull/2391

Glean currently generates the API using the application id. But the application id is different for different builds: Bzzzp!

**Comment by pocmo** [comment]
```
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (12, 26): Unresolved reference: debug
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (13, 26): Unresolved reference: debug
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (31, 13): Type inference for control flow expression failed. Please specify its type explicitly.
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (32, 44): Unresolved reference: Events
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (32, 64): Unresolved reference: Events
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (33, 50): Unresolved reference: Events
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (33, 76): Unresolved reference: Events
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (34, 45): Unresolved reference: Events
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (34, 66): Unresolved reference: Events
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (35, 50): Unresolved reference: Events
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (35, 76): Unresolved reference: Events
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (44, 17): Unresolved reference: apply
e: /opt/fenix/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt: (45, 13): Unresolved reference: defaultBrowser
```

**Comment by Dexterp37** [comment]
We are working on this!

**Comment by Dexterp37** [comment]
It's being fixed by mozilla-mobile/android-components#2507

**Comment by pocmo** [comment]
I'll trigger a new AC snapshot release now.

**Comment by vesta0** [comment]
Will be fixed by mozilla-mobile/android-components#2507

**Comment by boek** [comment]
We landed a fix in Fenix as well. Let me know if it works now!
